src/coreclr/jit/lower.cpp

```diff
@@ -357,10 +357,15 @@ GenTree* Lowering::LowerNode(GenTree* node)
         case GT_LCL_FLD_ADDR:
         case GT_LCL_VAR_ADDR:
         {
-            // TODO-Cleanup: this is definitely not the best place for this detection,
-            // but for now it is the easiest. Move it to morph.
+
             const GenTreeLclVarCommon* lclAddr = node->AsLclVarCommon();
-            comp->lvaSetVarDoNotEnregister(lclAddr->GetLclNum() DEBUGARG(Compiler::DNER_BlockOp));
+            const LclVarDsc*           varDsc  = comp->lvaGetDesc(lclAddr);
+            if (!varDsc->lvDoNotEnregister)
```

Comment:
Shouldn't this always call `lvaSetVarDoNotEnregister` and it can early-return if already set (but also possibly JitDump if already set for a different reason)

Reply:
not in this case in my opinion, if everything goes well it would be replaced with assert(varDsc->lvDoNotEnregister) soon because this code is a patch for some locals that are slipping morph analysis nowadays.

Reply:
not in this case in my opinion, if everything goes well it would be replaced with `assert(varDsc->lvDoNotEnregister)` soon because this code is a patch for some locals that are slipping morph analysis nowadays.